erts/emulator/beam/jit/arm/32/instr_fun.cpp

```diff
@@ -81,8 +81,75 @@ void BeamModuleAssembler::emit_i_make_fun3(const ArgLambda &Lambda,
 }
 
 void BeamGlobalAssembler::emit_apply_fun_shared() {
-    // TODO
-    ASSERT(false);
+    // TODO this is the first called emitter
+    Label finished = a.newLabel();
+
+    /* Put the arity and fun into the right registers for `call_fun`, and stash
+     * the argument list in ARG2 for the error path. We'll bump the arity as
+     * we go through the argument list. */
+    mov_imm(ARG3, 0);
+    a.ldr(ARG4, getXRef(0));
+    a.ldr(ARG2, getXRef(1));
+    {
+        Label unpack_next = a.newLabel(), malformed_list = a.newLabel(),
+              raise_error = a.newLabel();
+
+        auto x_register = getXRef(0);
+
+        ASSERT(x_register.shift() == 0);
+        x_register.setIndex(ARG3);
+        x_register.setShift(2);
+
+        a.mov(ARG1, ARG2);
+        a.bind(unpack_next);
+        {
+            a.cmp(ARG1, imm(NIL));
+            a.b_eq(finished);
+
+            ERTS_CT_ASSERT(_TAG_PRIMARY_MASK - TAG_PRIMARY_LIST == (1 << 1));
+            a.tst(ARG1, imm(1)),
+            a.b_ne(malformed_list),
```

Comment:
Why are there commas and not semicolon at the end of the two lines above?

Reply:
omg... just my erlang braincells adding chaos